llvm/lib/Target/RISCV/RISCVInstrInfoXCheri.td

```diff
@@ -961,26 +971,36 @@ class CheriSetCCPatGpcrNull<PatFrag CondOp, dag GprDag>
 class Swap<PatFrag BinFrag>
     : PatFrag<(ops node:$a, node:$b), (BinFrag $b, $a)>;
 
+// NB: Constants are wrapped as (ptradd (inttoptr 0) C) so the operand order
+// isn't canonicalised like for integers so we have to add extra patterns for
+// the commutative and inverse cases. Maybe we should support Constant:iFATPTR,
+// represent them as (inttoptr C), or just canonicalise the current form?
+
+def  : CheriSetCCPatGpcrGpcr<setlt, (SLT node:$rs1, node:$rs2)>;
+defm : CheriSetCCPatGpcrSimm12<setlt, (SLTI node:$rs1, node:$imm12)>;
+defm : CheriSetCCPatGpcrSimm12<Swap<setgt>, (SLTI node:$rs1, node:$imm12)>;
+def  : CheriSetCCPatGpcrGpcr<setult, (SLTU node:$rs1, node:$rs2)>;
+defm : CheriSetCCPatGpcrSimm12<setult, (SLTIU node:$rs1, node:$imm12)>;
+defm : CheriSetCCPatGpcrSimm12<Swap<setugt>, (SLTIU node:$rs1, node:$imm12)>;
```

Comment:
Hoisted these because that's what RISCVInstrInfo.td does (as setlt and setult map directly to their equivalent instructions, whereas the others require alternatives)

(and included the corresponding commuted forms that aren't needed for RISC-V as I think it's clearer to have those next to their equivalent should-be-canonical form)